bndtools.builder/src/org/bndtools/builder/classpath/BndContainerInitializer.java

```diff
@@ -342,12 +416,26 @@ static void calculateWorkspaceBundleAccessRules(Map<Project,List<IAccessRule>> p
                 return;
             }
             Parameters exportPkgs = new Parameters(mf.getMainAttributes().getValue(new Name(Constants.EXPORT_PACKAGE)));
-            List<IAccessRule> tmp = new LinkedList<IAccessRule>();
+            List<IAccessRule> exports = new LinkedList<IAccessRule>();
+            List<IAccessRule> privates = new LinkedList<IAccessRule>();
+            Packages contained = model.getContained();
             for (String exportPkg : exportPkgs.keySet()) {
-                String pathStr = exportPkg.replace('.', '/') + "/*";
-                tmp.add(JavaCore.newAccessRule(new Path(pathStr), IAccessRule.K_ACCESSIBLE));
+                String exportDir = exportPkg.replace('.', '/');
+                String pathStr = exportDir + "/*";
+                privates.add(JavaCore.newAccessRule(new Path(pathStr), IAccessRule.K_ACCESSIBLE));
+                for (PackageRef containedPackage : contained.keySet()) {
+                    String dir = containedPackage.getBinary();
+                    if (dir.equals(exportDir)) {
+                        exports.add(JavaCore.newAccessRule(new Path(pathStr), IAccessRule.K_ACCESSIBLE));
+                    }
+                }
+                //TODO -- fix bnd bug.
```

Comment:
I think so. Looks like copy and paste bug. The containsbinaryname method seems wrong. I didn't write anything up yet.